python/lsst/sims/catUtils/utils/SNIaLightCurveGenerator.py

```diff
@@ -56,6 +57,20 @@ def __init__(self, *args, **kwargs):
         self._brightness_name = 'flux'
         super(SNIaLightCurveGenerator, self).__init__(*args, **kwargs)
 
+    def light_curves_from_pointings(self, pointings, chunk_size=100000, lc_per_field=None):
+        if lc_per_field is not None:
+            warnings.warn("You have set lc_per_field in the SNIaLightCurveGenerator. "
+                          "This will limit the number of candidate galaxies queried from the "
+                          "CatSim database per field-of-view.  Because supernovae are randomly "
+                          "populated in those galaxies, there is no guarantee that the galaxies "
+                          "queried will have supernovae in them.  If the galaxies you actuall "
+                          "query do not host supernovae, you could get fewer light curves than "
+                          "you expect.")
+
```

Comment:
It may not be impossible, but it will take some significant design.  Ordinarily, lc_per_field is implemented as a SQL "LIMIT" on the database query.  For SNIa, we can't actually know how many light curves we got until after all of the processing has been run, so we would have to implement some kind of loop that checks the number of light curves returned and then does another database query (making sure not to get the same objects we got in the last query), repeating until the requested number of light curves have been generated.  Probably, this will need to be done by implementing new versions of most of the LightCurveGenerator methods in SNIaLightCurveGenerator.  I will open a ticket to do so in the future.

The ticket is SIM-2093